csaf/src/model/metadata.rs

```diff
@@ -3,7 +3,31 @@ use url::Url;
 
 #[derive(Clone, Debug, PartialEq, Eq, serde::Deserialize, serde::Serialize)]
 pub struct Distribution {
-    pub directory_url: Url,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub directory_url: Option<Url>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub rolie: Option<Rolie>,
+}
+#[derive(Clone, Debug, PartialEq, Eq, serde::Deserialize, serde::Serialize)]
+pub struct Rolie {
+    pub categories: Option<Vec<Url>>,
+    pub feeds: Vec<Feed>,
+    pub services: Option<Vec<Url>>,
```

Comment:
I think this needs a `#[serde(default)]` and `#[serde(skip…)]`. And I assume it can be collapsed into a single `Vec`?

Reply:
I do not understand this one.

Reply:
What's the difference of `None` vs `Some(vec![])`? If there is none, then that should just be a `Vec<>` IMHO.